### PR TITLE
feat(argo-image-updater): Update to use version 0.12

### PR DESCRIPTION
**Pull request by BeyerJC**
This is my first PR here so any tips are welcome.

https://github.com/argoproj-labs/argocd-image-updater/releases/tag/v0.12.0

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo).
* [x] My build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

Changes are automatically published when merged to `master`. They are not published on branches.

### CODE REVIEW

**Review by mkilchhofer — CHANGES_REQUESTED** [review]
Hi @BeyerJC Thanks for your contribution.

Looks like there is some more config incoming:
image

**Comment by BeyerJC** [comment]
@mkilchhofer Thanks for the hint :-) I just checked the release notes and totally forgot about checking the manifest.

File: charts/argocd-image-updater/templates/deployment.yaml [hunk 1]
```diff
@@ -9,6 +9,8 @@ spec:
   selector:
     matchLabels:
       {{- include "argocd-image-updater.selectorLabels" . | nindent 6 }}
+  strategy:
+    type: {{ .Values.deploymentStrategy | default "Recreate" }}
```
Comment (mkilchhofer) [inline charts/argocd-image-updater/templates/deployment.yaml:13]: Let's use what we implemented in the notifications chart:
- https://github.com/argoproj/argo-helm/blob/master/charts/argocd-notifications/values.yaml#L27
- https://github.com/argoproj/argo-helm/blob/master/charts/argocd-notifications/templates/deployment.yaml#L9

Comment (BeyerJC) [reply charts/argocd-image-updater/templates/deployment.yaml]: Good point ! I adjusted the strategy part and also updated the README with the helm-docs script.

File: charts/argocd-image-updater/values.yaml [hunk 2]
```diff
@@ -96,6 +99,14 @@ config:
     #   credentials: ext:/scripts/auth1.sh
     #   credsexpire: 10h
 
+  sshConfig:
+    {}
+    # data:
```
Comment (mkilchhofer) [inline charts/argocd-image-updater/values.yaml:104]: Can we drop the data in the structure here?

File: charts/argocd-image-updater/templates/configmap-sshconfig.yaml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+apiVersion: v1
```
Comment (mkilchhofer) [inline charts/argocd-image-updater/templates/configmap-sshconfig.yaml:1]: I know we use this structure in this repository but it looks a bit strange. WDYT about:

```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  labels:
    {{- include "argocd-image-updater.labels" . | nindent 4 }}
  name: argocd-image-updater-ssh-config
data:
  {{- with .Values.config.sshConfig }}
  {{- toYaml . | nindent 2 }}
  {{- end }}
```

In this chart theres already another configmap which implement a feature in this structure: `configmap-authscripts.yaml`

Comment (BeyerJC) [reply charts/argocd-image-updater/templates/configmap-sshconfig.yaml:1]: I also prefer to drop the data in the values.yaml. It felt also a bit strange to me, i copied that from another chart and thought that would be the prefered approach.

I will change that to your suggestion.

**Comment by BeyerJC** [comment]
@mkilchhofer Do you have an idea why the helm-doc check is failing ? I generated the Readme with the helm-docs script but somehow, checksums are still not in sync.

**Comment by BeyerJC** [comment]
Ok , got it. For some reason, helm-docs added an extra linebreak at line 3 in the README.

**Comment by mkilchhofer** [comment]
> Ok , got it. For some reason, helm-docs added an extra linebreak at line 3 in the README.

Hmmm yeah, I didn't have time to analyze that :-D this is strange. But it looks good now 😎

**Review by mkilchhofer — APPROVED** [review]

**Pull request merged**